### PR TITLE
added model processing file and endpoint in server file

### LINKED ISSUE #38: Backend - Process User Input and Return Model Response
Develop the backend functionality to handle incoming user input from the frontend, process it, and return a response generated by a model. For testing and demonstration purposes, we will use the Hugging Face model (Salesforce/codet5-base) (https://huggingface.co/Salesforce/codet5-base) as a mock model. This backend API will enable interaction between the frontend’s user input and the model's response, allowing the UI to simulate conversation with an AI model in a chat format.

Acceptance Criteria:

- Endpoint Setup:
  - Create an API endpoint that accepts a POST request with JSON data 
  - Validate the request to ensure the "text" field is present and non-empty.

- Process Input with Mock Model:
   - Use the Hugging Face model Salesforce/codet5-base as a mock to generate responses. This model will serve as a placeholder to test the system until the actual model is integrated.
   - Integrate the Hugging Face API or Transformers library to interact with the model and process the user’s input.

- Model Processing Script:
   - Set up a Python script using the transformers library that takes in the text input, runs it through the Salesforce/codet5-base model, and returns the generated output.(You can find the model script in the link with description)
   - This script will serve as a backend utility that can be called by the API to generate responses.

- Response Handling:
   - Return the model’s output in a JSON format  so the frontend can display it.
   - If there is any processing error, return an appropriate error message with a clear HTTP status code.

- Error Handling:
   - Handle possible errors such as invalid input, model processing failures, and server errors.
   - Provide meaningful error messages for each error type.

Additional Information:

- This setup is intended as a mock to test the chat interface and backend flow.
- Later, the mock model can be replaced with the actual bioinformatics model once it is ready for integration.

### PR DESCRIPTION
**Pull request by Omp06**
Fixes #38 

**What was changed?**

The backend functionality was added to handle user input on the Homescreen of the application. Along with handling the input, the functionality to process the input through the mock AI model from Hugging Face called Salesforce/codet5-base was implemented. This was done through an API endpoint to send and receiver the input and output. The model instantiation was also implemented to utilize the correct model and interact with it from Hugging Face. 

**Why was it changed?**

This change was implemented to allow the user to do more than just customization and login to the application. The backend API endpoint will enable interaction between the frontend’s user input and the model's response. This would allow the UI to simulate conversation with an AI model on the HomeScreen of the application. This also prepares the API for when we utilize a bioinformatics model in the future since we are currenlty using a mock model. 

**How was it changed?**

In the server.js file, a new POST endpoint was defined that accepts the JSON input from the frontend. The endpoint ensure the incoming text meets criteria and it creates a python process to run the model and capture its output in the required format. A model.py file was created. In that file, the Salesforce/codet5-base model is loaded in from Hugging Face utilizing the transformers library. This file reads the input text and processes it with the model and outputs a response in JSON format for the UI to display. 

**Screenshots that show the changes (if applicable):**

### CODE REVIEW

File: llm_bioinformatics_research/server.js [hunk 1]
```diff
@@ -166,6 +167,51 @@ app.post('/home', async (req, res) => {
     }
 });
 
+const {spawn} = require('child_process');
```
Comment (oam67) [inline llm_bioinformatics_research/server.js:170]: When I run npm start I get an error with exit code 1. The error is telling me about const {spawn} being defined earlier. I see that it is defined at line 10 of this file so I think deleting this repeat here should at least resolve the error I am seeing. I'll also add a screenshot of the error.
image

File: llm_bioinformatics_research/model.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Load the model and tokenizer
+tokenizer = AutoTokenizer.from_pretrained("Salesforce/codet5-base")
+model = AutoModelForSeq2SeqLM.from_pretrained("Salesforce/codet5-base")
+pipe = pipeline("text2text-generation", model=model, tokenizer=tokenizer)
+
+# Read input from stdin
+def main():
+    input_text = json.loads(sys.stdin.read()).get("input_text", "")
+    if input_text:
+        result = pipe(input_text, max_length=50, num_return_sequences=1)
```
Comment (oam67) [inline llm_bioinformatics_research/model.py:14]: Is the max_length parameter controlling the length of the model's output? Will this type of limit be something we keep in the future? I know chatgpt does something where if the output gets really long it asks the user if they want to continue generating the output. Maybe we could do something similar later down the road.

File: llm_bioinformatics_research/server.js [hunk 3]
```diff
@@ -166,6 +167,51 @@ app.post('/home', async (req, res) => {
...
+app.post('/home', (req, res) => {
+    const { text } = req.body;
+
+    // Validate that the text field is provided and not empty
+    if (!text || typeof text !== 'string' || text.trim() === '') {
+        return res.status(400).json({ error: "The 'text' field is required and must be non-empty." });
+    }
+
+    // Spawn a Python process to run the model script
+    const pythonProcess = spawn('python3', ['model.py']);
```
Comment (oam67) [inline llm_bioinformatics_research/server.js:181]: If we are using python3 to spawn the python process to run the model script, I think we should add to our documentation that python3 will be needed. This way users set up the right python version when setting up our application for usage.

File: llm_bioinformatics_research/server.js [hunk 4]
```diff
@@ -166,6 +167,51 @@ app.post('/home', async (req, res) => {
...
+
+    // Send the input text to the Python script
+    pythonProcess.stdin.write(JSON.stringify({ text }));
+    pythonProcess.stdin.end();
+
+    let output = '';
+
+    // Capture the output from the Python script
+    pythonProcess.stdout.on('data', (data) => {
+        output += data.toString();
```
Comment (oam67) [inline llm_bioinformatics_research/server.js:191]: I think saving the output of the model in the output variable is fine for now but I wonder if we would need another way to store output if the output becomes really large for complex tasks.

File: llm_bioinformatics_research/server.js (hunk 1 above)
Comment (AjithAkuthota23) [reply llm_bioinformatics_research/server.js:170]: we can remove line 170 because it is already declared in line 10

**Pull request merged**